services/payments/secrets.go

```diff
@@ -260,17 +399,36 @@ func (s *Service) decryptSecrets(ctx context.Context) (map[string]string, error)
 		return nil, fmt.Errorf("failed to parse private key bytes for secret decryption: %w", err)
 	}
 
-	buf := bytes.NewBuffer(s.secretsCiphertext)
+	return age.Decrypt(bytes.NewReader(buf.Bytes()), identity)
+}
 
-	r, err := age.Decrypt(buf, identity)
+func (s *Service) encryptWithShares(ctx context.Context, data []byte) (io.Reader, error) {
+	// combine the service configured key shares
+	privateKey, err := shamir.Combine(s.keyShares)
 	if err != nil {
-		return nil, fmt.Errorf("failed to decrypt the secrets ciphertext: %w", err)
+		return nil, fmt.Errorf("failed to combine keyShares: %w", err)
 	}
 
-	var output = map[string]string{}
-	if err := json.NewDecoder(r).Decode(&output); err != nil {
-		return nil, fmt.Errorf("failed to json decode the secrets: %w", err)
+	identity, err := age.ParseX25519Identity(string(privateKey))
+	// zero the private key
+	for i := 0; i < len(privateKey); i++ {
+		privateKey[i] = 0
+	}
```

Comment:
```suggestion
```

Let's just delete this and open a follow up issue here. It's a defense in depth measure and after investigating further into this, I don't like the solution that comes from https://pkg.go.dev/github.com/awnumar/memguard and the author of age and golang crypto lib maintainer suggests that achieving this in golang is extremely hard because of the GC.

I'm instead changing my view here after investigating further and believe what we're doing is fine as is and we'll address this separately once `runtime/secrets` becomes available as discussed here: https://github.com/golang/go/issues/21865#issuecomment-942575996